Dockerfile

```diff
@@ -13,6 +15,11 @@ RUN pip install --no-cache-dir --upgrade --prefix="/install" -r requirements.txt
 
 FROM python:3.10-slim-bullseye as final
 
+RUN apt-get update && apt-get install -y \
```

Comment:
do we need compilers in final image,
they have already served their purpose

Reply:
Yes, it is necessary, because the Docker script is currently configured to rebuild hnswlib on boot (which is necessary to make sure that hnswlib's capabilities match the actual hardware (e.g AVX/AVX2/AVX512).

Originally we only built hnswlib when building the docker image, but then if you ran it on a different kind of CPU from the one used to build it it could cause errors. This is a case where Docker is a leaky abstraction.